mantle/platform/api/aws/ec2.go

```diff
@@ -108,77 +108,97 @@ func (a *API) CreateInstances(name, keyname, userdata string, count uint64, minD
+		key := &keyname
+		if keyname == "" {
+			key = nil
+		}
+
+		var rootBlockDev []*ec2.BlockDeviceMapping
+		if minDiskSize > 0 {
+			rootBlockDev = append(rootBlockDev, &ec2.BlockDeviceMapping{
+				DeviceName: aws.String("/dev/xvda"),
+				Ebs: &ec2.EbsBlockDevice{
+					VolumeSize: &minDiskSize,
+				},
+			})
+		}
+		inst := ec2.RunInstancesInput{
+			ImageId:             &a.opts.AMI,
+			MinCount:            &cnt,
+			MaxCount:            &cnt,
+			KeyName:             key,
+			InstanceType:        &a.opts.InstanceType,
+			SecurityGroupIds:    []*string{&sgId},
+			SubnetId:            &subnetId,
+			UserData:            ud,
+			BlockDeviceMappings: rootBlockDev,
+			TagSpecifications: []*ec2.TagSpecification{
+				&ec2.TagSpecification{
+					ResourceType: aws.String(ec2.ResourceTypeInstance),
+					Tags: []*ec2.Tag{
+						&ec2.Tag{
+							Key:   aws.String("Name"),
+							Value: aws.String(name),
+						},
+						&ec2.Tag{
+							Key:   aws.String("CreatedBy"),
+							Value: aws.String("mantle"),
+						},
 					},
 				},
 			},
-		},
-	}
-	if useInstanceProfile {
-		inst.IamInstanceProfile = &ec2.IamInstanceProfileSpecification{
-			Name: &a.opts.IAMInstanceProfile,
 		}
-	}
+		if useInstanceProfile {
+			inst.IamInstanceProfile = &ec2.IamInstanceProfileSpecification{
+				Name: &a.opts.IAMInstanceProfile,
+			}
+		}
 
-	var reservations *ec2.Reservation
-	err = util.RetryConditional(5, 5*time.Second, func(err error) bool {
-		// due to AWS' eventual consistency despite ensuring that the IAM Instance
-		// Profile has been created it may not be available to ec2 yet.
-		if awsErr, ok := err.(awserr.Error); ok && (awsErr.Code() == "InvalidParameterValue" && strings.Contains(awsErr.Message(), "iamInstanceProfile.name")) {
-			return true
+		err = util.RetryConditional(5, 5*time.Second, func(err error) bool {
+			// due to AWS' eventual consistency despite ensuring that the IAM Instance
+			// Profile has been created it may not be available to ec2 yet.
+			if awsErr, ok := err.(awserr.Error); ok && (awsErr.Code() == "InvalidParameterValue" && strings.Contains(awsErr.Message(), "iamInstanceProfile.name")) {
+				return true
+			}
+			return false
+		}, func() error {
+			var ierr error
+			reservations, ierr = a.ec2.RunInstances(&inst)
+			return ierr
+		})
+		if err == nil {
+			// Successfully started our instance in the requested zone. Break out of the loop
+			break
+		}
+		if err != nil {
+			// Handle InsufficientInstanceCapacity error by looking for instances in other zones
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "InsufficientInstanceCapacity" {
+				// If we iterate over all possible zones and none of them have sufficient instance(s)
+				// available we will return the InsufficientInstanceCapacity error
+				if zoneKey == len(zones)-1 {
+					return nil, fmt.Errorf("all available zones tried: %v", err)
+				}
+				fmt.Printf("Insufficient instances available in zone %v. Trying the next zone\n", zone)
```

Comment:
Looks like we use `plog` for other messages in this file.

```suggestion
				plog.Infof("Insufficient instances available in zone %v. Trying the next zone\n", zone)
```

Can you verify this still shows up even if you run `kola` without `--debug` ?

Reply:
No, it doesn't show up without `--debug`

Reply:
Maybe we make it a `plog.Warnf` then? Does that show up?

Reply:
That does show up. The only thing is that shows up in the this format:
`2022-08-25T20:23:51Z platform/api/aws: Insufficient instances available in zone`